cpp/arcticdb/python/python_module.cpp

```diff
@@ -227,12 +191,33 @@ void register_error_code_ecosystem(py::module& m) {
     setattr(m, "enum_value_to_prefix", enum_value_to_prefix);
     m.def("get_error_category", &get_error_category);
 
-    py::register_exception<InternalException>(m, "InternalException");
-    py::register_exception<NormalizationException>(m, "NormalizationException");
-    py::register_exception<MissingDataException>(m, "MissingDataException");
-    py::register_exception<SchemaException>(m, "SchemaException");
-    py::register_exception<StorageException>(m, "StorageException");
-    py::register_exception<SortingException>(m, "SortingException");
+    // legacy exception base type kept for backwards compat with Man Python client
+    struct ArcticCompatibilityException : public std::runtime_error {};
+    auto compat_exception = py::register_exception<ArcticCompatibilityException>(
+            m, "_ArcticLegacyCompatibilityException");
+
+    py::register_exception<InternalException>(m, "InternalException", compat_exception.ptr());
```

Comment:
If you read the source code of `register_exception`, you'll find it's similar to calling `py::register_exception_translator` for `InternalException`, so the first `catch` below is redundant.

Reply:
No it isn't, it's necessary to stop it being caught by the `std::exception` catch.

Reply:
But the `register_exception_translator` on line 201 is called first?

That call will throw an `InternalException` either on line 205 or 211 and be caught again by the exception handler registered on line 199, which rethrows that exception to Python.

You can, however, remove the second rethrowing overhead by adapting this example from the pybind docs:

```cpp
static py::exception<MyCustomException> exc(m, "MyCustomError");

py::register_exception_translator([](std::exception_ptr p) {
    try {
        if (p) std::rethrow_exception(p);
    } catch (const MyCustomException &e) {
        exc(e.what());
...
```

Here, the `exc` is a variable pointing at the exception class in Python. Calling that will register an exception to be thrown when switching back to Python.